pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +136,37 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			gid := fileinfo.Sys().(*syscall.Stat_t).Gid
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			// If fsGroup volume ownership was set for the mounter path, set it for the sub path too
+			fsGroup := pod.Spec.SecurityContext.FSGroup
+			if gid == uint32(*fsGroup) {
+				if err := os.MkdirAll(hostPath, perm); err != nil {
+					glog.Errorf("failed to mkdir:%s", hostPath)
+					return nil, err
+				}
+				volume.SetVolumeOwnership(vol.Mounter, fsGroup, hostPath)
+
+				// If umask is preventing us from making the sub path have the same permissions as the
+				// mounter path, chmod the sub path
+				fileinfo, err := os.Lstat(hostPath)
```

Comment:
yes it's needed, SetVolumeOwnership() only ORs the perm bits by 0660/0440 (rw/ro). So we have to chmod too to make the sub path have the same perm bits as the mount path. E.g. an emptydir mount path has 0777 and the umask is 022, this has to happen to make sub path look the same